CMakeLists.txt

```diff
@@ -77,23 +70,37 @@ string(REGEX REPLACE " /W[0-4]" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
 # Duktape
 set(DUKTAPE_SOURCE_ROOT "" CACHE STRING "Root of the Duktape source tree")
-if (NOT DUKTAPE_SOURCE_ROOT)
-    message(FATAL_ERROR
-            "Duktape source root not found.\n"
-            "Please set DUKTAPE_SOURCE_ROOT to the root of the Duktape source tree.\n"
-            "https://github.com/svaarala/duktape/releases/download/v2.7.0/duktape-2.7.0.tar.xz\n")
+
+if (DUKTAPE_SOURCE_ROOT)
+    add_library(duktape ${DUKTAPE_SOURCE_ROOT}/src/duktape.c ${DUKTAPE_SOURCE_ROOT}/src/duktape.h)
+    target_include_directories(
+            duktape PUBLIC
+            "$<BUILD_INTERFACE:${DUKTAPE_SOURCE_ROOT}/src>"
+            "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>")
+
+    if (CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
+        target_compile_options(duktape PRIVATE -w)
+    elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        target_compile_options(duktape PRIVATE /w)
+    endif()
+    set(DUKTAPE_INSTALL duktape)
+else () # NOT DUKTAPE_SOURCE_ROOT
+    # try to find it on the machine
+    find_file(DUKTAPE_HEADER duktape.h)
+    find_library(DUKTAPE_LIBRARY duktape)
+    if (NOT DUKTAPE_HEADER STREQUAL DUKTAPE_HEADER-NOTFOUND
+        AND NOT DUKTAPE_LIBRARY STREQUAL DUKTAPE_LIBRARY-NOTFOUND)
+        add_library(duktape SHARED IMPORTED)
+        set_target_properties(duktape PROPERTIES IMPORTED_LOCATION ${DUKTAPE_LIBRARY})
+        find_library(DUKTAPE_LIBRARY duktape)
+    else()
```

Comment:
@alandefreitas how's that? Just import if it can be found, without any argument.s

Reply:
Yes. This looks OK. More idiomatic.

```
find_library(DUKTAPE_LIBRARY duktape)
```

is being called twice though. 

And the header path/directory is being ignored rather than set in the imported target.

Reply:
Ok, how about now?

Reply:
Good idea, done. You think it's ready to merge now?

Reply:
To be honest, I just reviewed the CMake scripts and the MRDOX_DECL/VISIBLE macros... I had a look at the rest of the PR and it looks OK to me but I don't have any serious informed opinion about it because I don't know what requirements Vinnie gave you.

It appears that he's busy now and I'm quite unsure about the state of authority over these decisions in the meanwhile. I've only been merging handlebars stuff in the meanwhile because it doesn't touch MrDox at all. We could wait for him or just ask him.